document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -233,13 +247,33 @@ public boolean updateSubDoc(Key key, String subDocPath, Document subDocument) {
   @Override
   public BulkUpdateResult bulkUpdateSubDocs(Map<Key, Map<String, Document>> documents)
       throws Exception {
+    Pair<Set<Key>, Long> updated = bulkUpdateSubDocsInternal(documents);
+    Set<Key> updatedDocs = updated.getLeft();
+    long partiallyUpdated = updated.getRight();
+    long fullyUpdated = updateLastModifiedTime(updatedDocs);
+    if (fullyUpdated != partiallyUpdated) {
+      LOGGER.error(
+          "Not all documents fully updated, documents fullyUpdated:{}, patiallyUpdated:{}, expected:{}",
+          fullyUpdated,
+          partiallyUpdated,
+          documents.size());
+    }
+    return new BulkUpdateResult(fullyUpdated);
+  }
+
+  private Pair<Set<Key>, Long> bulkUpdateSubDocsInternal(Map<Key, Map<String, Document>> documents)
```

Comment:
Can we introduce a private class to hold the return type?
By looking at the method signature, it is not clear what would be `Set<Key>` and `Long`.